smsdk/client.py

```diff
@@ -262,6 +262,24 @@ def get_kpis_for_asset(self, **kwargs):
         base_url = get_url(
             self.config["protocol"], self.tenant, self.config["site.domain"]
         )
+
+        # Get machine_types dataframe to check display name
+        machine_types_df = self.get_machine_types()
+        machine_types = []
+        # This loop is getting system name of machine_type if provided only display name in request.
+        for machine_type in kwargs["asset_selection"]["machine_type"]:
+            source_type = machine_types_df.loc[
+                machine_types_df["source_type_clean"] == machine_type, "source_type"
+            ]
+            if not source_type.empty:
+                source_type = list(set(source_type.to_list()))
+                machine_types.extend(source_type)
+            else:
+                machine_types.append(machine_type)
```

Comment:
Could this lookup be more efficient?  It strikes me the use of Pandas Dataframes here does not make the code easier to understand or more efficient.  I would prefer to deal with things are sets, or collections.

```python
# Convert display name / source type lookup table from dataframe
alias_tbl = machine_types_df.loc[
    :, ["source_type", "source_type_clean"]
].set_index("source_type_clean").groupby("source_type_clean")["source_type"].apply(set).to_dict()

# At this point my knowledge of pandas transforms fail.
machine_types = []
for machine_type in kwargs["asset_selection"]["machine_type"]:
    machine_types.extend(alias_tbl.get(machine_type, (machine_type,)))
```

Reply:
I've addressed your recommended changes in separate PR for a subtask here- https://github.com/sightmachine/sightmachine-sdk/pull/46